appshell/appshell_extensions_win.cpp

```diff
@@ -816,12 +817,100 @@ bool quickTestBufferForUTF8(UTFValidationState& validationState)
     return (GetBufferAsUTF8(validationState));
 }
 
+typedef std::map<std::string, long> CharSetMap;
 
-int32 ReadFile(ExtensionString filename, ExtensionString encoding, std::string& contents)
+// Mapping of CharSet to CodePage
+CharSetMap charSetMap =
 {
-    if (encoding != L"utf8")
-        return ERR_UNSUPPORTED_ENCODING;
+    // Below mappings are listed on website
+    // https://msdn.microsoft.com/en-us/library/windows/desktop/dd317756(v=vs.85).aspx
+    { "ISO-2022-JP", 50220 },
+    { "ISO-8859-6", 28596 },
+    { "ISO-2022-CN", 50227 },
+    { "ISO-2022-KR", 50225 },
+    { "GB18030", 54936 },
+    { "BIG5", 950 },
+    { "EUC-JP", 20932 },
+    { "euc-jp", 51932 },
+    { "EUC-KR", 949 },
+    { "ISO-8859-1", 28591 },
+    { "ISO-8859-2", 28592 },
+    { "ISO-8859-5", 28595 },
+    { "ISO-8859-6", 28596 },
+    { "ISO-8859-7", 28597 },
+    { "ISO-8859-8", 28598 },
+    { "ISO-8859-9", 28599 },
+    { "WINDOWS-1250", 1250 },
+    { "WINDOWS-1251", 1251 },
+    { "WINDOWS-1252", 1252 },
+    { "WINDOWS-1253", 1253 },
+    { "WINDOWS-1254", 1254 },
+    { "WINDOWS-1255", 1255 },
+    { "WINDOWS-1256", 1256 },
+    { "KOI8-R", 20866 },
+    { "IBM420", 420 },
+    { "IBM424", 424 },
+
+    // These are also supported 
+    { "ISO-8859-8-I", 38598 },
+    { "SHIFT_JIS", 932 },
+    { "ISO-8859-3", 28593 },
+    { "ISO-8859-4", 28594 },
+    { "UTF-16", 1200 },
+    { "UTF-16BE", 1201},
+    { "WINDOWS-1257", 1257 },
+    { "WINDOWS-1258", 1258 },
+    { "GB2312", 936 },
+    { "HZ-GB-2312", 52936 },
+    { "WINDOWS-874", 874 },
+    { "CP866", 866 },
+    { "IBM852", 852 }
```

Comment:
Is the charset map present in the other similar PR which was targeted for Linux?